std/bitmanip.d

```diff
@@ -886,6 +886,89 @@ public:
         return b;
     }
 
+    /**
+      Sets all the values in the $(D BitArray) to the
+      value specified by $(D val).
+     */
+    BitArray opSliceAssign(bool val)
+    {
+        for (size_t i = 0; i < _len; i++)
+            opIndexAssign(val, i);
+        return this;
+    }
+
+    ///
+    @system unittest
+    {
+        import std.algorithm.comparison : equal;
+
+        auto b = BitArray([1, 0, 1, 0, 1, 1]);
+
+        b[] = true;
+        // all bits are set
+        assert(b.bitsSet.equal([0, 1, 2, 3, 4, 5]));
+
+        b[] = false;
+        // none of the bits are set
+        assert(b.bitsSet.empty);
+    }
+
+    /**
+      Sets the bits of a slice of $(D BitArray) which
+      starts at the index $(D start) and ends at the
+      index ($D end) - 1 with the values specified by $(D val).
+     */
+    BitArray opSliceAssign(bool val, size_t start, size_t end)
+    in
+    {
+        assert(start >= 0 && start <= _len);
+        assert(end >=0 && end <= _len);
+        assert(start <= end);
+    }
+    body
+    {
+        for (size_t i = start; i < end; i++)
+            opIndexAssign(val, i);
```

Comment:
Yah, the bulk of this should assign one word at a time with care at the ends. @RazvanN7 I'm sure you'll have a good time with this.